lib/web/apiserver_test.go

```diff
@@ -980,9 +986,33 @@ func (s *WebSuite) TestWebAgentForward(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (s *WebSuite) TestRemoteProcessExitErrorGetsHandled(c *C) {
+	// Create a nonexistent system user.
+	pack := s.authPack(c, "foo", "baz")
+	ws, err := s.makeTerminal(pack, session.NewID())
+	c.Assert(err, IsNil)
+
+	defer ws.Close()
+
+	// Receive "Failed to launch: user: unknown user baz."
+	err = s.waitForRawEvent(ws, 5*time.Second)
+	c.Assert(err, IsNil)
+
+	// This seems to trigger reading of the next incoming message:
+	// "the connection was closed on the remote side on..."
+	websocket.Message.Send(ws, []byte{})
+	err = s.waitForRawEvent(ws, 5*time.Second)
+	c.Assert(err, IsNil)
+
+	// If process exits with error codes,
+	// no close event is sent and this will timeout.
+	err = s.waitForCloseEvent(ws, 5*time.Second)
```

Comment:
set a lower time out here (maybe 1s), since it's expected.
unit tests should generally last <1s.